types/react-query/index.d.ts

```diff
@@ -94,34 +114,52 @@ export type MutateFunction<TResults, TVariables extends object> = (
 ) => Promise<TResults>;
 
 export interface MutationResult<TResults> {
-    data: TResults | null;
-    isLoading: boolean;
+    status: 'idle' | 'loading' | 'error' | 'success'
+    data: TResults | undefined;
     error: null | Error;
     promise: Promise<TResults>;
-    reset: () => void;
 }
 
-export function setQueryData(
-    queryKey: string | [string, object],
-    data: any,
-    options?: {
-        shouldRefetch?: boolean
-    }
-): void | Promise<void>;
+export namespace queryCache {
 
-export function refetchQuery(
-    queryKey: string | [string, object] | [string, false],
-    force?: {
-        force?: boolean
-    }
-): Promise<void>;
+    export function prefetchQuery<TResult, TQueryKey extends QueryKey>(
+        queryKey: TQueryKey,
+        queryFn: QueryFunction<TResult, QueryFunctionArgs<TQueryKey>>,
+        options?: QueryOptions<TResult>
+    ): Promise<TResult>;
 
-export function refetchAllQueries(
-    options?: {
-        force?: boolean,
-        includeInactive: boolean
-    }
-): Promise<void>;
+    export function getQueryData<TResult>(
+        queryKey: QueryKey
+    ): TResult | undefined;
+
+    export function setQueryData<TResult>(
+        queryKey: QueryKey,
+        updater: TResult | ((oldData: TResult) => TResult)
+    ): void;
+
+    export function refetchQueries(
+        queryKeyOrPredicateFn: QueryKey | ((query: QueryKey) => boolean),
+        exact: boolean,
```

Comment:
most of these functions have optional params, i.e `exact?:` and `throwOnError?:`

Reply:
Indeed, will fix